servo/connectors/opsani_dev.py

```diff
@@ -214,6 +214,40 @@ async def check_kubernetes_service_type(self) -> None:
                 f"expected service type of ClusterIP or LoadBalancer but found {service.spec.type}"
             )
 
+    @servo.checks.require("Target container resource requests are within limits")
+    async def check_target_container_resources_within_limits(self) -> None:
+        # Load the Deployment
+        deployment = await servo.connectors.kubernetes.Deployment.read(
+            self.config.deployment,
+            self.config.namespace
+        )
+        # assert deployment, f"failed to read deployment '{self.config.deployment}' in namespace '{self.config.namespace}'"
+
+        # Find the target Container
+        target_container = None
+        for container in deployment.containers:
+            if container.name == self.config.container:
+                target_container = container
+                break
+
+        assert target_container, f"failed to find container '{self.config.container}' when verifying resource limits"
+
+        # Get resource requirements from container
+        container_cpu_request = servo.connectors.kubernetes.Millicore.parse(target_container.resources.requests["cpu"])
+        container_memory_request = servo.connectors.kubernetes.ShortByteSize.validate(target_container.resources.requests["memory"])
+        
+        # Get config values
+        config_cpu_min = self.config.cpu.min
+        config_cpu_max = self.config.cpu.max
+        config_memory_min = self.config.memory.min
+        config_memory_max = self.config.memory.min
+        
+        # Check values against config.
+        assert config_cpu_min <= container_cpu_request, f"target container requests '{container_cpu_request}' cpu but config specifies a minimum of '{config_cpu_min}'"
+        assert container_cpu_request <= config_cpu_max, f"target container requests '{container_cpu_request}' cpu but config specifies a maximum of '{config_cpu_max}'"
+        assert config_memory_min <= container_memory_request, f"target container requests '{container_memory_request}' memory but config specifies a minimum of '{config_memory_min}'"
+        assert container_memory_request <= config_memory_max, f"target container requests '{container_memory_request}' memory but config specifies a maximum of '{config_memory_max}'"
```

Comment:
Extra credit here would be to switch this into a multi-check.

The value is that when they are run all checks that are not `required` are executed:

In a single check case like this, if you screwed up the CPU and the memory settings you would get one error (CPU), go fix it, then get the other error (memory), and go fix it.

With a multi-check expansion, they will both get run and you will see a progression -- this is why all the checks are so short and atomic.